people_recognition_3d/scripts/people_recognition_3d_node

```diff
@@ -93,13 +96,31 @@ class PeopleRecognition3DNode:
             "[Service Call] Detecting people in 3D from incoming RGB-D image..."
         )
         try:
-            people3d, _, _ = self._people_recognizer_3d.recognize(
+            people3d, markers, regions_viz = self._people_recognizer_3d.recognize(
                 req.image_rgb, req.image_depth, req.camera_info_depth)
         except Exception as e:
             rospy.logerr(
                 "[Service Call] Exception while recognizing people in 3D")
             raise
         else:
+            def validate_marker(marker):
+                def is_valid_float(f):
+                    return -1e3 < f < 1e3
+
+                def is_valid_vector(v):
+                    return is_valid_float(v.x) and is_valid_float(v.y) and is_valid_float(v.z)
+
+                def is_valid_q(q):
+                    return q.x ** 2 + q.y ** 2 + q.z ** 2 + q.w ** 2 == 1.
```

Comment:
Comparing a float to be equal to 1 is very sensitive for errors

Reply:
It is being compared with a floating point number `1.`. Also the quaternion needs be a unit quaternion, so this is fine.

Reply:
In theory yes, if computers were infinitely precise.  Floating point math rounding errors might bite you here I'm afraid. I'd use `np.isclose` or something similar.

Reply:
I have removed that segment of the code as the line using it was commented. The function `validate_marker` was written as Rviz was crashing for some reason in the tournament. It was later found that the marker for pointing_pose was having an invalid quaternion, which was then commented out in `people_recognizer_3d` and the light saber part was added instead.